### PR TITLE
Conference

### PR DESCRIPTION
**Pull request by lazedo**

### CODE REVIEW

File: applications/ecallmgr/src/ecallmgr_conference_command.erl [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+            {<<"play">>, Arg}
+    end;
+
+%% The following conference commands can optionally specify a participant
+get_conf_command(<<"play">>, _Focus, ConferenceId, JObj) ->
+    case kapi_conference:play_v(JObj) of
+        'false' ->
+            {'error', <<"conference play failed to execute as JObj did not validate.">>};
+        'true' ->
+            UUID = kz_json:get_ne_value(<<"Call-ID">>, JObj, ConferenceId),
```
Comment (fenollp) [inline applications/ecallmgr/src/ecallmgr_conference_command.erl]: How about local functions to fetch this and Participant-ID too (or any fields fetched in more than one place)?
This way they would be handled with the same code.
Also, what about get_ne_binary_value in both cited examples?

File: applications/ecallmgr/src/ecallmgr_conference_control.erl [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    Server = props:get_value('server', Props),
+    ConferenceId = props:get_value('conference_id', Props),
+    Event = kz_json:get_value(<<"Event">>, JObj),
+    handle_conference_event(Node, Server, ConferenceId, Event, JObj).
+
+handle_conference_event(_Node, Server, ConferenceId, <<"conference-destroy">>, _JObj) ->
+    lager:debug("RECEIVED CONFERENCE-DESTROY FOR ~s", [ConferenceId]),
+    gen_server:stop(Server);
+handle_conference_event(_Node, _Server, ConferenceId, Event, JObj) ->
+    lager:debug("EVENNT ~s FOR ~s NOT HANDLED : ~p", [Event, ConferenceId, JObj]).
```
Comment (fenollp) [inline applications/ecallmgr/src/ecallmgr_conference_control.erl]: debug code?

File: applications/ecallmgr/src/ecallmgr_conference_control.erl [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+handle_conference_event(JObj, Props) ->
+    'true' = kapi_conference:event_v(JObj),
+    Node = props:get_value('node', Props),
+    Server = props:get_value('server', Props),
+    ConferenceId = props:get_value('conference_id', Props),
+    Event = kz_json:get_value(<<"Event">>, JObj),
+    handle_conference_event(Node, Server, ConferenceId, Event, JObj).
+
+handle_conference_event(_Node, Server, ConferenceId, <<"conference-destroy">>, _JObj) ->
+    lager:debug("RECEIVED CONFERENCE-DESTROY FOR ~s", [ConferenceId]),
```
Comment (fenollp) [inline applications/ecallmgr/src/ecallmgr_conference_control.erl]: debug code?

File: applications/ecallmgr/src/ecallmgr_conference_command.erl (hunk 1 above)
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_conference_command.erl]: this code was moved, not refactored

File: applications/ecallmgr/src/ecallmgr_conference_control.erl (hunk 2 above)
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_conference_control.erl]: yep, needs cleaning

File: applications/ecallmgr/src/ecallmgr_fs_conference.erl [hunk 4]
```diff
@@ -234,11 +248,14 @@ publish_event(Action, Props, Node) ->
...
 
-publish_event(Action, #conference{handling_locally='true'} = Conference, Props, _Node) ->
-    Event = conference_event(Action, Conference, Props),
-    publish_event(Event);
-publish_event(Action, _Conference, _Props, _Node) ->
-    lager:debug("conference control on another node, not publishing event ~s", [Action]).
+publish_event(Action, #conference{handling_locally=IsLocal} = Conference, Props, _Node) ->
+    case props:is_true(<<"Force-Publish-Event-State">>, Props, 'false')
+        orelse props:is_true(<<"Publish-Event-State">>, Props, 'true')
+        andalso IsLocal
```
Comment (fenollp) [inline applications/ecallmgr/src/ecallmgr_fs_conference.erl]: I understand this `andalso`'s left hand side only encloses `props:is_true(<<"Publish-Event-State">>, Props, 'true')`.
The `Force...` expression is thus the top-level expression: `A orelse B andalso C` has this precedence: `A orelse (B andalso C)`.

Is this expected? Should we add braces for clarity?

**Pull request merged**